pydantic_ai_slim/pydantic_ai/settings.py

```diff
@@ -70,3 +78,35 @@ def merge_model_settings(base: ModelSettings | None, overrides: ModelSettings |
         return base | overrides
     else:
         return base or overrides
+
+
+@dataclass
+class ExecutionLimitSettings:
+    """Settings to configure an agent run."""
+
+    request_limit: int | None = None
+    request_tokens_limit: int | None = None
+    response_tokens_limit: int | None = None
+    total_tokens_limit: int | None = None
+
+    _request_count: int = 0
+    _request_tokens_count: int = 0
+    _response_tokens_count: int = 0
+    _total_tokens_count: int = 0
```

Comment:
Should these be public?

Reply:
I don't like the idea that the settings object also holds state, it feels to me like there should be a separate object for tracking state, and we can check the state against the settings. If I were a user I'd be inclined to reuse an instance of `ExecutionLimitSettings` which obviously will cause issues.

I would imagine we make a private type `_UsageState` or similar (which holds all the fields you are talking about), and have one of `ExecutionLimits` and `_UsageState` have a method that accepts the other and raises an error if appropriate.

Reply:
If we want to put the usage state on the runcontext we can make it public, but I feel like we can do that later/separately. I'll note that I could imagine Samuel disagreeing with all this, and I wouldn't find that unreasonable.